contracts/Wrapper20.sol

```diff
@@ -30,14 +36,12 @@ contract Wrapper20 is IERC20 {
         override
         returns (bool)
     {
-        // TODO(nlordell): This can be implemented by keeping `allowances`
-        // separate from `approvals`. With this separate state, `transferFrom`
-        // can be relatively trivially implemented such that:
-        // - if the `operator` is the sender or is ERC1155 approved, the amount
-        // is transferred like an ERC20 `transfer`
-        // - otherwise the amount is tranferred like an ERC20 `transferFrom`,
-        // that is, the amount is deducted from the allowance.
-        revert("W: not implemented");
+        // NOTE: ERC20 allowances are fundamentally different from ERC1155
+        // approvals, so when an allowance is set, ensure that the any approval
+        // is removed.
+        wrappable.sudoUnsetApprovalForAll(msg.sender, spender, id);
```

Comment:
I find it weird that setting the allowance on one condition may unset my prior allowance on other conditions (and thus affects existing allowances on other ERC20 wrappers). On the other hand I understand that it would be weird and potentially against the ERC20 standard to set a specific allowance and then have `allowance` still return U256::max

What would happen if we consider the ERC20 allowance as separate (ignoring the ERC1155 allowance)? It would cost one extra approval for each ERC20 even if the ERC1155 has already been approved but that could be done via permit.

Reply:
That's the other option. Then we can completely ignore ERC1155 approvals altogether.

Reply:
I like this suggestion a lot, as ERC20 allowances and ERC1155 approvals really are fundamentally different and don't really work well together.